backend/src/main/java/com/revature/happyfarmersmarket/service/CartService.java

```diff
@@ -0,0 +1,153 @@
+package com.revature.happyfarmersmarket.service;
+import com.revature.happyfarmersmarket.dao.CartDAO;
+import com.revature.happyfarmersmarket.dao.CartItemDAO;
+import com.revature.happyfarmersmarket.dao.ProductDAO;
+import com.revature.happyfarmersmarket.dao.UserDAO;
+import com.revature.happyfarmersmarket.exception.APIException;
+import com.revature.happyfarmersmarket.exception.ResourceNotFoundException;
+import com.revature.happyfarmersmarket.interceptor.AuthUtil;
+import com.revature.happyfarmersmarket.interceptor.UserDetails;
+import com.revature.happyfarmersmarket.model.Cart;
+import com.revature.happyfarmersmarket.model.CartItem;
+import com.revature.happyfarmersmarket.model.Product;
+import com.revature.happyfarmersmarket.model.User;
+import com.revature.happyfarmersmarket.payload.CartDTO;
+import com.revature.happyfarmersmarket.payload.ProductDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.support.StandardServletMultipartResolver;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.modelmapper.ModelMapper;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@Service
+public class CartService {
+
+    private static final Logger logger = LogManager.getLogger();
+
+    @Autowired
+    private CartDAO cartDAO;
+    @Autowired
+    private ProductDAO productDAO;
+    @Autowired
+    private CartItemDAO cartItemDAO;
+    @Autowired
+    private AuthUtil authUtil;
+    @Autowired
+    private UserService userService;
+    @Autowired
+    private UserDAO userDAO;
+    @Autowired
+    private StandardServletMultipartResolver standardServletMultipartResolver;
```

Comment:
Standard practice is to inject these via the constructor and make them final.

Reply:
Done!